File: Measurement Studio/Examples/DotNET/v4.5/Applications/Application/UISnippets/vb/Control.vb
Public Class SnipsControl
    Private control As Control
    Private lastBoxItem As Object
    Private randNumGenerator As Random

    Public Sub New(ByVal control As Control)
        Me.control = control
        randNumGenerator = New Random(DateTime.Now.Millisecond)
    End Sub

    ''' <summary>
    ''' A random number generator belonging to all
    ''' snips control types.
    ''' </summary>
    Protected ReadOnly Property RandNumberGenerator() As Random
        Get
            Return randNumGenerator
        End Get
    End Property

    ''' <summary>
    ''' This property retuns a reference to the control that
    ''' is being encapsulated by the SnipsControl instance
    ''' </summary>
    Public ReadOnly Property InternalControl() As Control
        Get
            Return control
        End Get
    End Property

    ''' <summary>
    ''' Reset the control to its default state
    ''' </summary>
    Public Overridable Sub ResetToDefaultState()
    End Sub

    Public Property LastComboBoxItem() As Object
        Get
            Return lastBoxItem
        End Get
        Set(ByVal value As Object)
            lastBoxItem = value
        End Set
    End Property
End Class

File: Measurement Studio/Examples/DotNET/v4.5/Applications/Application/UISnippets/cs/Control.cs
using System;
using System.Windows.Forms;

namespace NationalInstruments.Examples.Snippets
{
    /// <summary>
    /// This is the base Snips class from which all other derive.
    /// </summary>
    public class SnipsControl
    {
        private Control control;
        private object lastComboBoxItem;
        private Random randNumGenerator;

        /// <summary>
        /// A random number generator belonging to all
        /// snips control types.
        /// </summary>
        protected Random RandNumberGenerator { get { return randNumGenerator; } }

        /// <summary>
        /// Public contstructor for Snips control
        /// </summary>
        /// <param name="control">Base control instance</param>
        public SnipsControl(Control control)
        {
            this.control = control;
            randNumGenerator = new Random(DateTime.Now.Millisecond);
        }

        /// <summary>
        /// This property retuns a reference to the control that
        /// is being encapsulated by the SnipsControl instance
        /// </summary>
        public Control InternalControl { get { return control; } }

        /// <summary>
        /// Reset the control to its default state
        /// </summary>
        public virtual void ResetToDefaultState() { }

        /// <summary>
        /// The last combobox item to be selected
        /// </summary>
        public object LastComboBoxItem
        {
            get { return lastComboBoxItem; }
            set { lastComboBoxItem = value; }
        }
    }
}
